examples/acoustic/example_constant.py

```diff
@@ -60,34 +57,31 @@ def run(dimensions=(50, 50, 50), spacing=(20.0, 20.0, 20.0), tn=1000.0,
     location[0, 0] = origin[0] + dimensions[0] * spacing[0] * 0.5
     location[0, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
     location[0, 2] = origin[1] + 2 * spacing[2]
-    src.set_receiver_pos(location)
-    src.set_shape(nt, 1)
-    src.set_traces(time_series)
+    src = PointSource(name='src', data=time_series, coordinates=location)
 
     # Receiver geometry
     receiver_coords = np.zeros((101, 3))
     receiver_coords[:, 0] = np.linspace(0, origin[0] +
                                         dimensions[0] * spacing[0], num=101)
     receiver_coords[:, 1] = origin[1] + dimensions[1] * spacing[1] * 0.5
     receiver_coords[:, 2] = location[0, 1]
-    data.set_receiver_pos(receiver_coords)
-    data.set_shape(nt, 101)
+    rec = Receiver(name='rec', ntime=nt, coordinates=receiver_coords)
 
-    Acoustic = Acoustic_cg(model, data, src, nbpml=nbpml, t_order=time_order,
-                           s_order=space_order, dse=dse, dle=dle)
+    solver = AcousticWaveSolver(model, source=src, receiver=rec,
+                                time_order=time_order, space_order=space_order)
 
     info("Applying Forward")
-    rec, u, gflopss, oi, timings = Acoustic.Forward(save=full_run, dse='basic', dle=dle)
+    rec, u, summary = solver.forward(save=full_run, dse='basic', dle=dle)
 
     if not full_run:
-        return gflopss, oi, timings, [rec, u.data]
+        return summary.gflopss, summary.oi, summary.timings, [rec, u.data]
 
     info("Applying Adjoint")
-    Acoustic.Adjoint(rec, dse=dse, dle=dle)
+    solver.adjoint(rec, dse=dse, dle=dle)
     info("Applying Born")
-    Acoustic.Born(dm, dse=None, dle=dle)
+    solver.born(dm, dse=None, dle=dle)
     info("Applying Gradient")
-    Acoustic.Gradient(rec, u, dse=dse, dle=dle)
+    solver.gradient(rec.data, u, dse=dse, dle=dle)
```

Comment:
Why do you sometimes need the `.data` , and some other times (`solver.` above) the object?